File: backend/src/main/java/pl/aleokaz/backend/post/Comment.java
package pl.aleokaz.backend.post;

import pl.aleokaz.backend.user.User;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;
import java.util.Set;
import java.util.UUID;

@Entity
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class Comment extends Interaction {
    @NonNull
    @ManyToOne(optional = false)
    @EqualsAndHashCode.Exclude
    @ToString.Exclude
    private Interaction parent;

    @Builder
    public Comment(
            UUID id,
            @NonNull Interaction parent,
            @NonNull String content,
            @NonNull Date createdAt,
            Date editedAt,
            @NonNull User author,
            Set<Reaction> reactions,
            Set<Comment> comments) {
        super(id, content, createdAt, editedAt, author, reactions, comments);
        this.parent = parent;
    }
}

File: frontend/lib/models/data/comment.dart
import 'package:ale_okaz/models/data/reactions.dart';

class Comment {
  final String id;
  final String content;
  final DateTime createdAt;
  final DateTime? editedAt;
  final String authorId;
  final Reactions reactions;
  final List<Comment> comments;

  Comment({
    required this.id,
    required this.content,
    required this.createdAt,
    this.editedAt,
    required this.authorId,
    required this.reactions,
    required this.comments,
  });

  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      id: json['id'] as String,
      content: json['content'] as String,
      createdAt: DateTime.parse(json['createdAt'] as String),
      editedAt: json['editedAt'] != null
          ? DateTime.parse(json['editedAt'] as String)
          : null,
      authorId: json['authorId'] as String,
      reactions: Reactions.fromJson(json['reactions'] as Map<String, dynamic>),
      comments: (json['comments'] as List)
          .map((e) => Comment.fromJson(e as Map<String, dynamic>))
          .toList(),
    );
  }
}
